asf_heat_pump_suitability/utils/geo_utils.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import geopandas as gpd
+import shapely
+
+
+def transform_gdf_drop_duplicates(gdf: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    """
+    Drop polygons with the same representative point. This drops both duplicate and nearly identical geometries.
+
+    Args:
+        gdf (gpd.GeoDataFrame): GeoDataFrame with polygon geometries
+
+    Returns:
+        gpd.GeoDataFrame: GeoDataFrame with duplicate polygon geometries dropped
+    """
+    # TODO: this assumes duplicate polygons will always generate the same representative points. Is this correct?
+    gdf["rep_point"] = gdf.representative_point()
```

Comment:
There's a margin speed up if you cast this to a well known binary instead of leaving it as a shapely point object:
```python
gdf["rep_point"] = gdf.representative_point().to_wkb()
```